qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/QuteCodeActionForUnknownMethod.java

```diff
@@ -49,23 +53,69 @@ public void doCodeActions(CodeActionRequest request, List<CompletableFuture<Void
 			if (node == null) {
 				return;
 			}
-			
+
 			ResolvedJavaTypeInfo baseResolvedType = request.getJavaTypeOfCoveredNode(javaCache);
 			if (baseResolvedType == null) {
 				return;
 			}
-			
+
 			MethodPart part = (MethodPart) node;
 			Template template = request.getTemplate();
 			Diagnostic diagnostic = request.getDiagnostic();
 
-			
-			// TODO : implement code actions
+			doCodeActionsForSimilarValues(part, template, diagnostic, baseResolvedType, codeActions);
 
 		} catch (BadLocationException e) {
 			LOGGER.log(Level.SEVERE, "Creation of unknown method code action failed", e);
 		}
 
 	}
 
+	/**
+	 * Create CodeAction(s) for similar text suggestions for methods
+	 *
+	 * @param part             the method part
+	 * @param template         the Qute template
+	 * @param diagnostic       the UndefinedVariable diagnostic
+	 * @param baseResolvedType the resolved Java type info
+	 * @param codeActions      list of CodeActions
+	 *
+	 * @throws BadLocationException
+	 */
+	private void doCodeActionsForSimilarValues(MethodPart part, Template template, Diagnostic diagnostic,
+			ResolvedJavaTypeInfo baseResolvedType, List<CodeAction> codeActions) throws BadLocationException {
+		Collection<String> availableValues = collectAvailableValuesForMethodPart(template, baseResolvedType);
+		doCodeActionsForSimilarValues(part, availableValues, template, diagnostic, codeActions);
+	}
+
+	/**
+	 * Collect similar text suggestions for methods
+	 *
+	 * @param template         the Qute template
+	 * @param baseResolvedType the resolved Java type info
+	 *
+	 * @return a list of available methods for the resolved type
+	 */
+	private Collection<String> collectAvailableValuesForMethodPart(Template template,
+			ResolvedJavaTypeInfo baseResolvedType) {
+		Collection<String> availableValues = new HashSet<>();
+
+		// Resolve methods defined in Java type
+		List<String> javaMethodNames = baseResolvedType.getMethods().stream().map(x -> x.getMethodName())
+				.collect(Collectors.toList());
+		for (String methodName : javaMethodNames) {
+			availableValues.add(methodName);
+		}
+

```

Comment:
You should take care of extended type too. For instance toString() which comes from Object doesn't appear as code action. Please write test too with this usecase.